### PR TITLE
Add jsonschema as a list format

### PR DESCRIPTION
**Pull request by Jeffail**

### CODE REVIEW

**Review by rockwotj — APPROVED** [review]
LGTM - thank you!

File: internal/jsonschema/stream_schema.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	defs := map[string]any{
+		"input":      compSpecsToDefinition(env.InputDocs(), docs.ReservedFieldsByType(docs.TypeInput)),
+		"buffer":     compSpecsToDefinition(env.BufferDocs(), docs.ReservedFieldsByType(docs.TypeBuffer)),
+		"cache":      compSpecsToDefinition(env.CacheDocs(), docs.ReservedFieldsByType(docs.TypeCache)),
+		"processor":  compSpecsToDefinition(env.ProcessorDocs(), docs.ReservedFieldsByType(docs.TypeProcessor)),
+		"rate_limit": compSpecsToDefinition(env.RateLimitDocs(), docs.ReservedFieldsByType(docs.TypeRateLimit)),
+		"output":     compSpecsToDefinition(env.OutputDocs(), docs.ReservedFieldsByType(docs.TypeOutput)),
+		"metrics":    compSpecsToDefinition(env.MetricsDocs(), docs.ReservedFieldsByType(docs.TypeMetrics)),
+		"tracer":     compSpecsToDefinition(env.TracersDocs(), docs.ReservedFieldsByType(docs.TypeTracer)),
+		"scanner":    compSpecsToDefinition(env.ScannerDocs(), docs.ReservedFieldsByType(docs.TypeScanner)),
```
Comment (mihaitodor) [inline internal/jsonschema/stream_schema.go:52]: Missing logger?

**Review by mihaitodor — COMMENTED** [review]
LGTM, just one small question.

File: internal/jsonschema/stream_schema.go (hunk 1 above)
Comment (Jeffail) [reply internal/jsonschema/stream_schema.go:52]: Not in that section, these are the lists of all plugin implementations so nothing to add for loggers. The config section is the `properties` bit below.

**Pull request merged**